src/main/java/org/stellar/sdk/SequenceNumberStrategy.java

```diff
@@ -0,0 +1,26 @@
+package org.stellar.sdk;
+
+public interface SequenceNumberStrategy {
+
+    /**
+     * Derives the desired sequence number for a newly built transaction.
+     *
+     * @param account the source account of transaction.
+     * @return the sequence number to be applied on new transaction.
+     */
+    long getSequenceNumber(TransactionBuilderAccount account);
```

Comment:
Interesting idea :thinking: I'm curious if @leighmcculloch thinks this is in line with how payment channels would want to and/or expect to use this

Is `getNextSequenceNumber` more accurate, to imply that this is generally an impure callback?

Reply:
historically we have tried to keep the API of the different SDKs as consistent as possible. Unless we plan to adopt a similar interface in the Go SDK I'd advise from deviating from making any changes to the API which significantly deviate from the Go SDK

Reply:
I refactored the tx builder sequence number resolution to be functional and pretty close to the same interface that txnbuild provides of either 'derive from sourceAccount.seqNum incremented by 1' or 'use an explicit value'.

https://github.com/stellar/java-stellar-sdk/pull/415/files#diff-338b40b80a77385ce8f94bb8c79269a588b87a112a370760184837e2a5345357R110

Reply:
I've [reverted back to original tx builder interface](https://github.com/stellar/java-stellar-sdk/pull/415/commits/4c8f5bf7d7152724b234d25c5b9d6e15a6c1e982) for using source account to get sequence number, just to move the p19 support through. we can do enhanced sequence number interfaces as follow-on if needed for tx's related to offchain payment flow use cases.